File: programming_languages/cpp/data_structures_implementation/queue/linearQueue.cpp
#include <iostream>
using namespace std;

class queue{

	int size, value, *a;
	int front, rear;

	public:
		queue( ){
			front = rear = -1;
			cout << "ENTER SIZE\t";
			cin >> size;

			a = new int [size];
		}
		void enqueue( void ){
			if( rear == size -1 ){
                cout << "OVERFLOW" << endl;
                return;
			}

			cout << "ENTER VALUE\t";
            cin >> value;
			if( front == -1 && rear == -1 ){
                front = rear = 0;
			}else{
                ++rear;
			}
            a[rear] = value;

		}
		void dequeue( void ){
            if( front == -1 || front > rear ){
                cout << "UNDERFLOW" << endl;
            }
            else{
                cout << a[front] << " DELETED" << endl;
                ++front;
            }
		}
		void display( void ){
            if( front <= rear ){
                for( int i = front; i <= rear; ++i)
                    cout << "a[" << i + 1 << "] = " << a[i] << endl;
            }
            else{
                cout << "EMPTY" << endl;
            }

		}
};

int main() {

	queue q;

	int choice;

	while( true ){

		cout << "1). ENQUEUE" << endl;
		cout << "2). DEQUEUE" << endl;
		cout << "3). DISPLAY" << endl;
		cout << "4). EXIT" << endl;
		cout << "ENTER CHOICE\t";
		cin >> choice;

		switch( choice ){
			case 1:
				q.enqueue();
				break;
			case 2:
				q.dequeue();
				break;
			case 3:
				q.display();
				break;
            case 4:
                return 0;
			default:
				cout << "ERROR CHOICE" << endl;
		}
	}

	return 0;
}

File: programming_languages/python/data_structures_implementation/queue/linear_queue.py
class queue( object ):
	def __init__(self):
		self.front = -1
		self.rear = -1
		self.size = int(input("Enter size of Queue:\t"))
		self.arr = [None]*self.size

	def enqueue(self):
		if self.rear == self.size-1:
			print("OVERFLOW")
		else:
			val = int(input("Enter value to be inserted\t"))

			if self.rear == -1 and self.front == -1 :
				self.front = 0
				self.rear = 0
			else:
				self.rear += 1

			self.arr[self.rear] = val
			print(val , "INSERTED")

	def dequeue(self):
		if self.front == -1 or self.front > self.rear:
			print("UNDERFLOW")
		else:
			print( self.arr[self.front] , "deleted")
			self.front+=1

	def display(self):
		if self.front <= self.rear:
			high = self.front
			low = self.rear + 1
			for x in range(high,low):
				print(self.arr[x])
		else:
			print ("EMPTY")

def controller():
	user_queue = queue()
	choice = None
	again = True

	while( again ):
		print("1). ENQUEUE")
		print("2). DEQUEUE")
		print("3). DISPLAY")
		print("4). EXIT")
		choice = int(input("ENTER CHOICE\t"))

		if choice == 1:
			user_queue.enqueue()
		elif choice == 2:
			user_queue.dequeue()
		elif choice == 3:
			user_queue.display()
		elif choice == 4:
			print("GOODBYE")
			again = None
		else:
			print("ERROR CHOICE")

controller()